packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -195,6 +196,25 @@ const DataGrid = forwardRef((props, ref) => {
   } = props;
 
   const displayedColumns = (hasSelectableRows ? [WorklistDataGridUtils.ROW_SELECTION_COLUMN] : []).concat(pinnedColumns).concat(overflowColumns);
+  // Create new displayedColumns object to account for column spans
+  const displayedColumnsWithColumnSpan = useMemo(() => {
+    let i = 0;
+    const newDisplayedColumns = [];
+    displayedColumns.forEach((column) => {
+      if (column.columnSpan > 1) {
+        newDisplayedColumns[i] = { ...column, columnSpanIndex: 0 };
+        i += 1;
+        for (let counter = column.columnSpan; counter > 1; counter -= 1) {
+          newDisplayedColumns[i] = { id: `${column.id}`, columnSpanIndex: `${column.columnSpan - counter + 1}` };
+          i += 1;
```

Comment:
I believe this should all be handled by a single for loop.  You should not have to manually set the first element.